src/file/ShareDialog.tsx

```diff
@@ -1,17 +1,92 @@
-import { IModalProps, IStyleFunctionOrObject, Theme } from '@fluentui/react';
-import React from 'react';
+import {
+    DialogFooter,
+    IModalProps,
+    IStyle,
+    IStyleFunctionOrObject,
+    Label,
+    Pivot,
+    PivotItem,
+    PrimaryButton,
+    Spinner,
+    TextField,
+    Theme,
+    mergeStyleSets,
+} from '@fluentui/react';
+import React, { useEffect, useState } from 'react';
 import { BaseDialog, IBaseDialogStyles } from '../BaseDialog';
+import { useScene } from '../SceneProvider';
+import { sceneToText } from '../file';
+
+const classNames = mergeStyleSets({
+    tab: {
+        minHeight: 200,
+        display: 'grid',
+        gridTemplateRows: '1fr auto',
+        gridTemplateAreas: `
+            "content"
+            "footer"
+        `,
+    } as IStyle,
+    footer: {
+        gridArea: 'footer',
+    } as IStyle,
+});
+
+const copyIconProps = { iconName: 'Copy' };
 
 export const ShareDialog: React.FC<IModalProps> = (props) => {
     return (
         <BaseDialog headerText="Share" {...props} dialogStyles={dialogStyles}>
-            <p>TODO</p>
+            <Pivot>
+                <PivotItem headerText="Export String" className={classNames.tab}>
+                    <ShareText />
+                </PivotItem>
+            </Pivot>
         </BaseDialog>
     );
 };
 
 const dialogStyles: IStyleFunctionOrObject<Theme, IBaseDialogStyles> = {
     body: {
+        minWidth: 500,
         maxWidth: '70ch',
     },
 };
+
+const labelStyles = mergeStyleSets({
+    message: { transition: 'none', opacity: 1 } as IStyle,
+    hidden: { transition: 'opacity 0.5s ease-in-out', opacity: 0 } as IStyle,
+});
+
+const ShareText: React.FC = () => {
+    const { scene } = useScene();
+    const [data, setData] = useState<string | undefined>(undefined);
+    const [copyMessageVisible, setMessageVisibility] = useState(false);
+    const [timeout, setMessageTimeout] = useState<number>();
+
+    useEffect(() => {
+        (async () => setData(await sceneToText(scene)))();
+    }, [scene]);
```

Comment:
I think you could simplify this a little bit using `useAsync()` from `react-async`. See the use in FileDialog.tsx OpenLocalFile for a good example of something that watches a value and asynchronously calculates something from it.

I think it would look something like this:
```ts
const { data, isPending } = useAsync(async () => sceneToText(scene), { watch: scene });
```

Reply:
Turns out `useAsync` and `useEffect` are no longer necessary - When I originally wrote `sceneToText` and `textToScene` they were async, however this changed later, but I forgot to update this section of the code. Now I can simply just `const data = sceneToText(scene);`

Reply:
I should mention - My current changes still does include `useEffect`, to ensure that the scene is only compressed once when the component is loaded.

Reply:
In that case, I think you want to use `useMemo`. This would only call `sceneToText()` when the component is first loaded and any time `scene` changes:

```ts
const data = useMemo(() => sceneToText(scene), [scene]);
```

Reply:
Ah that's correct - I've switched to useMemo and included it in this PR.